pkg/cloud/aws/services/kubeadm/aws_defaults.go

```diff
@@ -162,6 +165,9 @@ func SetJoinNodeConfigurationOverrides(caCertHash, bootstrapToken string, machin
 		klog.Infof("Overriding node's cloud-provider to the required value of %q.", cloudProvider)
 	}
 	base.NodeRegistration.KubeletExtraArgs["cloud-provider"] = cloudProvider
+	if util.IsControlPlaneMachine(machine.Machine) {
```

Comment:
It looks like the logic here is reversed, shouldn't this apply the nodeRole if it is not a control plane node?

```suggestion
	if !util.IsControlPlaneMachine(machine.Machine) {
```